lib/crypto/src/signature/error.rs

```diff
@@ -0,0 +1,46 @@
+//! Signature error type.
+
+use core::fmt::{self, Debug, Display};
+
+// TODO: Once we update `alloy` to a crates.io version, use the `signature`
+// crate directly for better composability with the ecosystem.
+/// Result type.
+///
+/// A result with the `signature` crate's [`Error`] type.
+pub type Result<T> = core::result::Result<T, Error>;
+
+/// Signature errors.
+///
+/// This type is deliberately opaque as to avoid sidechannel leakage which
+/// could potentially be used recover signing private keys or forge signatures
+/// (e.g. [BB'06]).
+///
+/// When the `std` feature is enabled, it impls [`std::error::Error`] and
+/// supports an optional [`std::error::Error::source`], which can be used by
+/// things like remote signers (e.g. HSM, KMS) to report I/O or auth errors.
+///
+/// [BB'06]: https://en.wikipedia.org/wiki/Daniel_Bleichenbacher
+#[derive(Default)]
+#[non_exhaustive]
+pub struct Error {}
+
+impl Error {
+    /// Create a new error with no associated source
+    pub fn new() -> Self {
```

Comment:
Should we add some explanation in error what happened wrong during signature check?

Reply:
These are opaque errors, we don't return any information that may enable a sidechannel attack.

It's explained in line 14 of that file, in the documentation for the struct.

```rust
/// Signature errors.
///
/// This type is deliberately opaque as to avoid sidechannel leakage which
/// could potentially be used recover signing private keys or forge signatures
/// (e.g. [BB'06]).
///
/// [BB'06]: https://en.wikipedia.org/wiki/Daniel_Bleichenbacher
```